File: 0680: validPalindromeII.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s)-1
        def thinIceValid(x, y):
            while x < y:
                if s[x] == s[y]:
                    x+=1
                    y-=1
                else:
                    return False
            return True
        while i < j:
            if s[i] == s[j]:
                i+=1
                j-=1
            else:
                return thinIceValid(i+1,j) or thinIceValid(i,j-1)
        return True
        

File: 0680: validPalindromeII.cpp
class Solution {
public:
    bool validPalindrome(string s) {
        
        int i = 0;
        int j = s.size()-1;
        while(i < j){

            if(s[i] != s[j]){

                return thinIce(s, i+1, j) || thinIce(s, i, j-1);
            }

            i++;
            j--;
        }

        return true;
    }

    bool thinIce(string s, int i, int j){

        while(i < j){

            if(s[i] != s[j]){

                return false;
            }

            i++;
            j--;
        }

        return true;
    }
};